InputNumber: NumberFormatException when symbol=" " (Whitespace)

**Describe the defect**
We have some use cases, where the `symbol` Property of an InputNumber is produced by String concatenation. In some cases this results in a Symbol only consisting of a blank (Whitespace) character.
Whenever that happens, the InputNumber produces erroneous number outputs, which can't be deserialized in the backend, because they contain whitespaces in a bad place.

**Reproducer**
https://github.com/metaxmx/primefaces-test/tree/reproducer-inputnumber-trailing-space

**Environment:**
 - PF Version: 10.0.0 (also tested latest master with 11.0.0-SNAPSHOT)
 - JSF + version: Mojarra 2.3
 - Affected browsers: Firefox

**To Reproduce**
Steps to reproduce the behavior:
1. Open Reproducer page in Browser
2. Select all text in input and overwrite by pressing the "5" key
3. Click outside the input filed to trigger the blur of the input, so that the new value will be sent to backend
4. See Error:
```
javax.faces.FacesException: Invalid number
	at org.primefaces.component.inputnumber.InputNumberRenderer.decode(InputNumberRenderer.java:94)
	at javax.faces.component.UIComponentBase.decode(UIComponentBase.java:509)
	at javax.faces.component.UIInput.decode(UIInput.java:798)
	at javax.faces.component.UIComponentBase.processDecodes(UIComponentBase.java:886)
	at javax.faces.component.UIInput.processDecodes(UIInput.java:692)
	at com.sun.faces.context.PartialViewContextImpl$PhaseAwareVisitCallback.visit(PartialViewContextImpl.java:606)
       ...
Caused by: java.lang.NumberFormatException: Character   is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:582)
	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:467)
	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:896)
	at org.primefaces.component.inputnumber.InputNumberRenderer.decode(InputNumberRenderer.java:89)

```

**Expected behavior**
The correct number "5.00" should be sent to the backend, and the success FacesMessage should be shown.

**Example XHTML**
```html
<h:form id="frmTest">
    <p:growl id="grwl"/>
    <p:inputNumber id="number" symbol=" " symbolPosition="s" value="#{testView.decimal}">
        <p:ajax update="@form" process="@this"/>
    </p:inputNumber>
</h:form>
```

**Example Bean**
```java
@Named
@ViewScoped
public class TestView implements Serializable {
    
    private BigDecimal decimal = new BigDecimal("3.4");

    public BigDecimal getDecimal() {
        return decimal;
    }

    public void setDecimal(BigDecimal decimal) {
        this.decimal = decimal;
        FacesContext.getCurrentInstance().addMessage(null, new FacesMessage("New value: " + getDecimal()));
    }

}
```

I have tracked this down to a Bug in Autonumeric (https://github.com/autoNumeric/autoNumeric/issues/721) in combination with a blur event in `1-inputnumber.js`:
```js
    bindInputEvents: function() {
        var $this = this;

        // GitHub #6447: browser auto fill fix
        this.input.off('blur.inputnumber').on('blur.inputnumber', function(e) {
            var element = AutoNumeric.getAutoNumericElement(this);
            if (element && this.value && this.value.length > 0) {
                var newValue = this.value;
                if($this.cfg.digitGroupSeparator) {
                    newValue = newValue.replaceAll($this.cfg.digitGroupSeparator, '');
                }
                element.set(newValue, null, true);
            }
            $this.copyValueToHiddenInput();
        });
    },
```
In the case, that `symbol=" "`, this will result in calling `element.set(newValue, null, true);` where `newValue` contains trailing whitespaces (e.g. `element.set('5 ', null, true);`.
As i described in AutoNumeric Issue https://github.com/autoNumeric/autoNumeric/issues/721, this will case the autonumeric to return e.g. `5 `, when we call `var val = this.autonumeric.getNumericString();` in the `getValue()` function, which will then result in a broken number string.

I think we have multiple ways of dealing with this (apart from waiting for the AutoNumeric bug to be fixed):
We could correct the returned value in the `getValue()` function, by adding a `trim()` to it (if it is a string).
But since the additional whitespace seems to lead to strange behaviours in the AutoNumeric itself (especially if the whitespace is added *before* the number), I think we should avoid adding the whitespaces to AutoNumeric in the first place by trimming the value in `bindInputEvents()`.

--- a/primefaces/src/main/resources/META-INF/resources/primefaces/inputnumber/1-inputnumber.js
+++ b/primefaces/src/main/resources/META-INF/resources/primefaces/inputnumber/1-inputnumber.js
@@ -146,7 +146,7 @@ PrimeFaces.widget.InputNumber = PrimeFaces.widget.BaseWidget.extend({
         this.input.off('blur.inputnumber').on('blur.inputnumber', function(e) {
             var element = AutoNumeric.getAutoNumericElement(this);
             if (element && this.value && this.value.length > 0) {
-                var newValue = this.value;
+                var newValue = this.value.trim();
                 if($this.cfg.digitGroupSeparator) {
                     newValue = newValue.replaceAll($this.cfg.digitGroupSeparator, '');
                 }

